File: benchmarks/correctness/python/reverse_complement.py
"""Reverse complement all sequences from FASTA. Output JSON."""
import json
import sys
from Bio import SeqIO
from Bio.Seq import Seq

results = {}
for record in SeqIO.parse("data/contigs.fa", "fasta"):
    rc = str(Seq(str(record.seq)).reverse_complement())
    results[record.id] = rc

json.dump({"sequences": results}, sys.stdout, indent=2)
print()

File: benchmarks/correctness/r/reverse_complement.R
# Reverse complement all sequences from FASTA. Output JSON.
library(Biostrings)
library(jsonlite)

seqs <- readDNAStringSet("data/sequences.fa")
# readDNAStringSet keeps the entire header line; BioLang and BioPython
# use the identifier up to the first space.
names(seqs) <- sub(" .*$", "", names(seqs))
results <- sapply(names(seqs), function(nm) {
    as.character(reverseComplement(seqs[[nm]]))
})

cat(toJSON(list(sequences = as.list(results)), auto_unbox = TRUE, pretty = TRUE, digits = 10))
cat("\n")
